packages/test/test-service-load/runloadtest.ps1

```diff
@@ -80,34 +83,78 @@ function CreateAndUploadConfig{
 		[Parameter(Mandatory = $true)]
         [string]$StorageAccountName,
 		[Parameter(Mandatory = $true)]
-        [string]$StorageAccountKey
+        [string]$StorageAccountKey,
+        [Parameter(Mandatory = $true)]
+        [AllowEmptyString()]
+        [string]$TenantDocsFilePath
     )
 
     $Tenants = (Get-Content -Raw -Path testTenantConfig.json | ConvertFrom-Json).tenants
+    $TenantDocs = $null
+    # case 1 : If TenantDocsFilePath is not provided then PODS need to create their own config file
+    # case 2 : If TenantDocsFilePath is provided
+    #   case a : If TenantDocsFilePath path exists.In that case, we read that file :
+    #       case a:  If we read file content and it doesn't have tenantDocUrls object then we throw error and return.
+    #       case b : Proceed ahead for populating url in config file. 
+    #   case b : If path doesn't exist then we throw error and exit.          
+    if($TenantDocsFilePath -ne ''){
+        if((Test-Path $TenantDocsFilePath)){
+            $TenantDocs = (Get-Content -Raw -Path $TenantDocsFilePath | ConvertFrom-Json).tenantDocUrls
+            if(!$TenantDocs){
+                Write-Error "File content of $TenantDocsFilePath is corrupted.Exiting..."
+                return
+            }
+        }else{
+            Write-Error "$TenantDocsFilePath file does not exist.Exiting..."
+            return
+        }   
+    }     
     $TenantNames = $Tenants | Get-Member -MemberType NoteProperty | Select-Object -ExpandProperty Name
     $TenantsCount = $TenantNames.Count
 	$Pods = $(kubectl get pods -n $Namespace --field-selector status.phase=Running -o json | ConvertFrom-Json).items
     [int]$PodsCount = $Pods.count
 
     if ( $PodsCount -ne $NumOfDocs ) {
-        Write-Error "Number of pods not equal to number of docs"
-        return
-     }
-
+       Write-Error "Number of pods not equal to number of docs"
+       return
+    }
     Write-Output "Starting pod's configuration creation at $(Get-Date)"
+    #Hast table is maintained for storing index of url per tenant which can be used next time
+    $TenantUrlIndexHt = [hashtable]::new()
+    for($i = 0; $i -lt $TenantNames.length; $i++){
+        $TenantUrlIndexHt[$TenantNames[$i]] = 0
+    } 
     $TempPath = "\tmp"
 	az storage directory create --account-key $StorageAccountKey  --account-name $StorageAccountName  --name $TestUid --share-name fluid-config-store
 	$PodConfigPath = (Join-Path -Path $TempPath -ChildPath $TestUid)
 	New-Item -Path $PodConfigPath -ItemType Directory | out-null
 	foreach ($i in 1..$PodsCount) {
 		$PodName = $Pods[$i - 1].metadata.name
         $TenantIndex = ($i-1) % $TenantsCount
-		$TenantId = $TenantNames[$TenantIndex]
-		$TenantContent = [PSCustomObject]@{
+        $TenantId = $TenantNames[$TenantIndex]
```

Comment:
is this correct indentation?

Reply:
Showing in the review board like this but in file it is correct.
Don't know why it is showing like this here.